### ISSUE TITLE
Consider iFrame Title Filter

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#3393]
## Issue

SiteImprove and other a11y tools have has previously flagged our embeds for not having a "title." Since we don't directly control the iframe markup this has been out of reach.

## Proposed Solution

- [x] Audit our current embed options for title attributes and see if there are SiteImprove issues (local extension).
- [x] Install and configure https://www.drupal.org/project/iframe_title_filter to see if it remedies any of the embeds that are missing title attributes and see if SiteImprove (local extension) no longer flags the embed for this issue.

**Comment by richardbporter** [comment]
If #3838 is the route we take, we could ensure a title is set in that filter.

**Comment by joewhitsitt** [comment]
I think this should be re-opened. with #3838 we are only allowing a few sources and that functionality does not extend to our oEmbed objects.

This just came up in the SiteImprove presentation today

**Comment by joewhitsitt** [comment]
Facebook, Twitter Kalutura, YouTube and Vimeo provide a title attribute.

UICapture (Panopto), Instagram do not.

**Comment by briand44** [comment]
>Facebook, Twitter Kalutura, YouTube and Vimeo provide a title attribute.
UICapture (Panopto), Instagram do not.

Verify this and try out the module listed in the proposed solution.

**Comment by joewhitsitt** [comment]
I think this will only go so far though and may still be flagged. If I recall, SiteImprove is flagging the inner iframe object which we have no control over.

YouTube - note that the outer iframe does have a title attribute, but the inner does not:
<img width="739" alt="Screen Shot 2022-05-10 at 1 27 54 PM" src="https://user-images.githubusercontent.com/4663676/167697528-f1b0c557-00aa-4621-a5ca-5eacee7660db.png">

**Comment by makurj** [comment]
Installed the module but there are no configuration options to complete this step: "Under "Filter processing order," make sure that this filter is set to run after any HTML filtering, as well as after any filters that would be generating iFrames (e.g., video_filter)."

Here's what I see with whatever the default settings for that module are: No title.
ScheduleOnce:
![image](https://user-images.githubusercontent.com/52135878/168377604-9b6f36ee-ea67-46fa-a2a9-287d9fe41453.png)

YouTube:
![image](https://user-images.githubusercontent.com/52135878/168377502-2656a74f-58d6-4dfb-92a7-be41af6a76dc.png)


Also found: https://www.drupal.org/project/drupal/issues/3085545

**Comment by joewhitsitt** [comment]
@makurj this module should present itself as an additional WYSIWYG filter that can be enabled and ordered here: admin/config/content/formats/manage/filtered_html

As I mentioned in the comment above, this might not solve our YouTube SiteImprove issue (inner iframe), but it is still something we should consider for Panopto and Instagram if they are still an issue.

**Comment by makurj** [comment]
Also found: https://www.drupal.org/project/iframe_title_filter/issues/3211492

**Comment by makurj** [comment]
Is this the expected behavior for a Panopto video? It's not picking up the title I gave it when I uploaded the video through the media library, "Knowledge Share".

![image](https://user-images.githubusercontent.com/52135878/168386018-0c8b6492-3ef5-4aed-8916-6c15222da786.png)

**Comment by makurj** [comment]
If I embed a Panopto video using the embed code from the video and using Full HTML -> source, it removes the title:

![image](https://user-images.githubusercontent.com/52135878/168386724-310f3417-7907-43e7-aebf-3ea14ceaccd1.png)

**Comment by joewhitsitt** [comment]
I don't know what steps you have done. Based on similar solutions, this is what I believe might be the recipe for success:

- composer require drupal/iframe_title_filter
- sync sandbox
- go to /media/media-panopto and see the iframe object there. there isn't a title attribute on render.
- enable the module
- go to admin/config/content/formats/manage/filtered_html
- enable the filter. save.
- there is a place to change the order of what fires first. make sure the module fires after media. save.
- cache rebuild.
- check /media/media-panopto and check for the title attribute. if there is one, I think it would be worth exporting the config and starting a PR. check instagram. try a new panopto (UI Capture) through media. there should be no full-html work involved with this.
- like i said before, i doubt this alone will solve the youtube issue, but it helps elsewhere.

**Comment by makurj** [comment]
Yep, did all the steps except for the /media/media-panopto one, but on default. Will continue working on it.

**Comment by joewhitsitt** [comment]
"Embedded content from uicapture.hosted.panopto.com" looks correct to me. That is what the module is supposed to do.

https://github.com/uiowa/uiowa/issues/3393#issuecomment-1126471294

https://git.drupalcode.org/project/iframe_title_filter/-/blob/8.x-1.x/src/Plugin/Filter/FilteriFrameTitle.php#L36

**Comment by makurj** [comment]
No iframe object here: /admin/structure/media/manage/panopto

**Comment by makurj** [comment]
Did not work for Instagram or Google Maps:

![image](https://user-images.githubusercontent.com/52135878/168656101-02f14e55-01c1-45d8-878c-7851167c3c97.png)

![image](https://user-images.githubusercontent.com/52135878/168656110-a94dbb7a-1f42-4aa8-9478-9d20b54c5131.png)

**Comment by makurj** [comment]
Candid Career seems to work:

![image](https://user-images.githubusercontent.com/52135878/168657051-2e86c3b3-ea09-4a27-ae75-732e9d7df06d.png)

**Comment by joewhitsitt** [comment]
>Candid Career seems to work

in that the iframe has a title attribute and the module does not try and replace it?

**Comment by makurj** [comment]
I see no difference in the DOM for the same video here https://careers.uiowa.edu/career-communities/helping-counseling#explore-helping-and-counseling-careers so I'm not sure what, if anything, is being done by the module in this case.
![image](https://user-images.githubusercontent.com/52135878/168657858-bcefdba9-59f0-410e-97ef-a1bab7e37205.png)

**Comment by makurj** [comment]
Pass: Panopto, Candid Career.
Fail: ScheduleOnce, YouTube, Instagram and Google Maps.

![image](https://user-images.githubusercontent.com/52135878/168658432-0ea9c8bf-2597-4de0-a682-22f5330a0306.png)
